src/org/opensolaris/opengrok/configuration/Project.java

```diff
@@ -215,16 +215,15 @@ public static Project getProject(File file) {
      * @return project that fits the name
      */
     public static Project getByName(String name) {
-        Project ret = null;
         RuntimeEnvironment env = RuntimeEnvironment.getInstance();
         if (env.hasProjects()) {
             for (Project proj : env.getProjects()) {
-                if (name.indexOf(proj.getName()) == 0) {
-                    ret = proj;
+                if (name.equals(proj.getName())) {
```

Comment:
Actually if there is this change, the container type can be finally changed from `List` to `Set (TreeSet)` which will cause ultra speedup for this lookup.